serde-generate/tests/dart_runtime.rs

```diff
@@ -14,13 +15,56 @@ fn install_test_dependency(path: &Path) -> Result<()> {
 
     Ok(())
 }
+
 #[test]
 fn test_dart_runtime() {
     let tempdir = tempdir().unwrap();
     let source_path = tempdir.path().join("dart_project");
 
     let registry = test_utils::get_registry().unwrap();
```

Comment:
The way I understand this test it seems to be superficially using the Rust module `test_utils` to generate some format then import (templated) Dart code from `runtime/dart/test/` and verify that deserialization work in the generated code.

The problem that I have with this approach is that

1. It is completely different from what we have in other languages so IMHO it's creating more maintenance burden than needed just for Dart

2. What happens if I want to change `get_registry()` later? Currently, this impacts only tests written in Rust with quoted code in other languages, and most of the format-dependent logics is shared between languages. Here we have to maintain tight consistency with a Dart templated file in `runtime/dart`.

3. This approach does not test interoperability with Rust bytes as the other xxx_runtime.rs files do

4. It cannot be easily extended to test the many corner cases of (de)serialization as we did in other languages.

Reply:
@matbd thanks for the feedback, that makes sense. I had somehow missed (!) that the `reference.map()` code in other runtime tests was injecting the Rust bytes into the runtime code and so I was focused on just "testing the runtime". I've pushed up an implementation that's almost identical to Typescript/Go/etc now if you want to take a look.